frontend/test/pytest/test_conditionals.py

```diff
@@ -664,5 +665,107 @@ def branch_f():
         assert func(False) == 0
 
 
+class TestCondPredicateConversion:
```

Comment:
Let's also confirm what the error message is when the conversion with `astype` does fail, and make sure we think it's appropriate enough for the user to understand and fix.

Reply:
Done!